### PR TITLE
feat: add VERS semver versioning scheme support

### LINKED ISSUE #56: feat: add VERS semver versioning scheme support
## Problem Statement

The VERS specification supports generic SemVer versioning scheme, but go-univers currently only implements SemVer ecosystem support without VERS integration. Users cannot use VERS range notation with SemVer versions like `vers:semver/>=1.2.0|<2.0.0`.

## Research

From the [VERS specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst), SemVer follows "MAJOR.MINOR.PATCH format" as defined in the Semantic Versioning Specification 2.0.0. The [univers Python implementation](https://github.com/aboutcode-org/univers) includes generic SemVer support in its VERS implementation.

SemVer provides a pure implementation of Semantic Versioning 2.0.0 without ecosystem-specific extensions.

## Current State

- ✅ SemVer ecosystem exists at `pkg/ecosystem/semver/`
- ✅ SemVer version parsing and comparison implemented following SemVer 2.0.0 spec
- ❌ No VERS integration for SemVer versioning scheme
- ❌ Cannot parse/evaluate ranges like `vers:semver/>=1.2.0|<2.0.0`

## Implementation Strategy

Following the established pattern used for npm, pypi, go, and maven VERS schemes:

1. **Create `pkg/spec/vers/semver.go`** with:
   - `semverContains(constraints []string, version string) (bool, error)` function
   - `intervalToSemverRanges(interval interval) []string` function

2. **Update `pkg/spec/vers/vers.go`**:
   - Add `"semver": semverContains` to `schemeToContains` map
   - Add `case "semver": rangeStrs = intervalToSemverRanges(interval)` to `toRanges` function
   - Update supported ecosystems documentation

3. **Add comprehensive tests** in `pkg/spec/vers/semver_test.go`

## Success Criteria

- [ ] Support VERS SemVer ranges: `vers:semver/>=1.2.0`
- [ ] Handle prerelease versions correctly: `vers:semver/>=1.0.0-alpha.1|<2.0.0`
- [ ] Handle metadata versions: `vers:semver/>=1.0.0+build123`
- [ ] Support all VERS operators: `>=`, `<=`, `>`, `<`, `=`, `!=`
- [ ] Pass comprehensive test suite covering edge cases
- [ ] CLI integration: `univers vers contains "vers:semver/>=1.2.0" "1.5.0"`
- [ ] Consistent behavior with existing SemVer ecosystem implementation

## Resources

- [VERS Specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst)
- [Univers Python Implementation](https://github.com/aboutcode-org/univers)
- [Semantic Versioning 2.0.0 Specification](https://semver.org/)
- Existing implementations: [npm VERS](https://github.com/alowayed/go-univers/issues/35), [pypi VERS](https://github.com/alowayed/go-univers/issues/36), [go VERS](https://github.com/alowayed/go-univers/issues/37)
- Pattern reference: `pkg/spec/vers/npm.go`

### PR DESCRIPTION
**Pull request by alowayed**
## Summary

Implements VERS (Version Range Specification) support for the generic SemVer ecosystem, enabling users to use VERS range notation with SemVer versions like `vers:semver/>=1.2.0|<2.0.0`.

- **Add `semverContains()` function** for generic SemVer VERS constraint checking
- **Add `intervalToSemverRanges()` function** to convert intervals to semver range syntax  
- **Update vers.go** to include semver in `schemeToContains` map and `toRanges` switch
- **Comprehensive test suite** covering all VERS operators and edge cases
- **CLI integration** with command `univers vers contains "vers:semver/>=1.2.0" "1.5.0"`

## Implementation Details

Following the established pattern used for npm, pypi, go, maven, gem, cargo, and alpine VERS schemes:
- Created `pkg/spec/vers/semver.go` with semver-specific VERS functions
- Updated `pkg/spec/vers/vers.go` to register semver support
- Added comprehensive tests in `pkg/spec/vers/semver_test.go`

Key implementation insight: SemVer supports both space-separated and comma-separated constraints. We chose space-separated (like npm) for consistency and simplicity, generating ranges like `>=1.0.0 <=2.0.0`.

## Test Plan

- [x] All existing tests pass (no regressions)
- [x] New comprehensive test suite with 29 test cases
- [x] Code quality checks pass (golangci-lint)
- [x] CLI functionality verified

### Success Criteria Verification

- [x] **Support VERS SemVer ranges**: `vers:semver/>=1.2.0` → ✅
- [x] **Handle prerelease versions**: `vers:semver/>=1.0.0-alpha.1|<2.0.0` → ✅  
- [x] **Handle metadata versions**: `vers:semver/>=1.0.0+build123` → ✅
- [x] **Support all VERS operators**: `>=`, `<=`, `>`, `<`, `=`, `!=` → ✅
- [x] **Pass comprehensive test suite** covering edge cases → ✅
- [x] **CLI integration**: `univers vers contains "vers:semver/>=1.2.0" "1.5.0"` → ✅
- [x] **Consistent behavior** with existing SemVer ecosystem implementation → ✅

## Examples

```bash
# Basic range
univers vers contains "vers:semver/>=1.2.0|<=2.0.0" "1.5.0"  # → true

# Prerelease support  
univers vers contains "vers:semver/>=1.0.0-alpha.1|<2.0.0" "1.5.0"  # → true

# Build metadata support
univers vers contains "vers:semver/>=1.0.0+build123" "1.5.0+other"  # → true

# Exclusion operator
univers vers contains "vers:semver/>=1.0.0|!=1.5.0" "1.5.0"  # → false

# All operators supported
univers vers contains "vers:semver/>1.0.0|<2.0.0" "1.5.0"  # → true

# Complex constraints
univers vers contains "vers:semver/>=1.0.0-alpha|<=2.0.0-beta+build" "1.5.0"  # → true
```

## SemVer 2.0.0 Features

This implementation provides full Semantic Versioning 2.0.0 support:
- **Standard versioning**: `MAJOR.MINOR.PATCH` format
- **Prerelease versions**: `1.0.0-alpha.1`, `1.0.0-beta.2`
- **Build metadata**: `1.0.0+build123` (ignored in comparisons per SemVer spec)
- **Pure SemVer**: No ecosystem-specific extensions, following the official specification exactly

Fixes #56

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: pkg/spec/vers/semver.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"strings"
+
+	"github.com/alowayed/go-univers/pkg/ecosystem/semver"
+)
+
+// semverContains implements VERS constraint checking for generic SemVer ecosystem
+func semverContains(constraints []string, version string) (bool, error) {
+	e := &semver.Ecosystem{}
+	return contains(e, constraints, version)
+}
```
[the remark this answers is not in the document]
Comment (alowayed) [reply pkg/spec/vers/semver.go:14]: No. This adds a global variable.
